src/algorithms/fardetectors/MatrixTransferStatic.cc

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2022 Alex Jentsch, Wouter Deconinck, Sylvester Joosten, David Lawrence
+//
+// This converted from: https://eicweb.phy.anl.gov/EIC/juggler/-/blob/master/JugReco/src/components/FarForwardParticles.cpp
+
+#include "MatrixTransferStatic.h"
+
+void eicrecon::MatrixTransferStatic::init(std::shared_ptr<spdlog::logger> &logger) {
+
+  m_log = logger;
+
+  // do not get the layer/sector ID if no readout class provided
+  if (m_cfg.readout.empty()){m_log->error("READOUT IS EMPTY!"); return;}
+
+  //Calculate inverse of static transfer matrix
+  std::vector<std::vector<double>> aX(m_cfg.aX);
+  std::vector<std::vector<double>> aY(m_cfg.aY);
+
+  double det = aX[0][0] * aX[1][1] - aX[0][1] * aX[1][0];
+
+  if (det == 0) {
+    m_log->error("Reco matrix determinant = 0! Matrix cannot be inverted! Double-check matrix!");
+    return;
+  }
+
+  aXinv[0][0] =  aX[1][1] / det;
+  aXinv[0][1] = -aX[0][1] / det;
+  aXinv[1][0] = -aX[1][0] / det;
+  aXinv[1][1] =  aX[0][0] / det;
+
+
+  det = aY[0][0] * aY[1][1] - aY[0][1] * aY[1][0];
+
+  if (det == 0) {
+    m_log->error("Reco matrix determinant = 0! Matrix cannot be inverted! Double-check matrix!");
+    return;
+  }
+
+  aYinv[0][0] =  aY[1][1] / det;
+  aYinv[0][1] = -aY[0][1] / det;
+  aYinv[1][0] = -aY[1][0] / det;
+  aYinv[1][1] =  aY[0][0] / det;
+
+  return;
+
+}
+
+std::unique_ptr<edm4eic::ReconstructedParticleCollection> eicrecon::MatrixTransferStatic::produce(const edm4hep::SimTrackerHitCollection& rawhits) {
+
+  auto outputParticles = std::make_unique<edm4eic::ReconstructedParticleCollection>();
+
+  //---- begin Reconstruction code ----
+
+  double goodHitX[2] = {0.0, 0.0};
+  double goodHitY[2] = {0.0, 0.0};
+  double goodHitZ[2] = {0.0, 0.0};
+
+  bool goodHit1 = false;
+  bool goodHit2 = false;
+
+  for (const auto &h: rawhits) {
+
+    auto cellID = h.getCellID();
+    // The actual hit position in Global Coordinates
+    auto gpos = m_cellid_converter->position(cellID);
+    // local positions
+    auto volman = m_detector->volumeManager();
+    auto local = volman.lookupDetElement(cellID);
+
+    auto pos0 = local.nominal().worldToLocal(dd4hep::Position(gpos.x(), gpos.y(), gpos.z())); // hit position in local coordinates
+
+    if(!goodHit2 && gpos.z()/dd4hep::mm > m_cfg.hit2minZ && gpos.z()/dd4hep::mm < m_cfg.hit2maxZ){
```

Comment:
This should be done with a plane in the cellID, not a z range.

Reply:
I agree that this should make proper use of the geometry but I would rather leave any implementation changes for another PR even if they don't effect the functionality. I believe @ajentsch is working on updated versions of this.
Specifically for this comment, the current readout structure for the Off Momentum and Roman Pots in the simulation could do with an overhaul to converge, the Off Momentum layer is always "1".